packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -39,6 +49,17 @@ class FlatTermSelector extends Component {
 		};
 	}
 
+	unescapeTerm( term ) {
```

Comment:
Why is `unescapeTerm` deserving of a utility function, but the inverse `escapeTerm` is effectively performed inline in `findOrCreateTerm`. Should we be consistent? Can this function be documented?

Reply:
Order-wise, I'd prefer if we keep a convention of `constructor`, lifecycle, utility, `render` in components (i.e. move these utility functions under `componentDidMount` and friends).

Maybe worth enforcing via lint rules?

Reply:
I don't have a strong sense one way or the other, but it's unclear to me that this is something specific to the component, vs. a generalized function which could be defined outside the component definition (also allows for easier unit testing).

I might also be thinking this way in consideration of how we might want to have this data be managed in the future. Ideally we don't have to keep this bizarre escape dancing in this component itself, but rather allow it to be managed somewhere in `@wordpress/core-data`. At that point, we might still have `unescapeTerm`, but it certainly wouldn't be component-specific.

Reply:
> Why is unescapeTerm deserving of a utility function, but the inverse escapeTerm is effectively performed inline in findOrCreateTerm. Should we be consistent?

We don't have an escape function that escapes term objects; we escape the term name.  The function to escape the term name is just the function to escape strings in lodash. I thought about having escapeTermName and unescapeTermName functions, but they would be just an import as: `escape as escapeTermName,`and later if we need other escapes, it stops making sense.
Off course we can have escapeTerm escape the term and get the name after but that seems like an unnecessary waste of resources as we just need the escaped name and we will not save the escaped term anywhere.


> Order-wise, I'd prefer if we keep a convention of constructor, lifecycle, utility, render in components (i.e. move these utility functions under componentDidMount and friends).

> Maybe worth enforcing via lint rules?

Having a lint rule for that would be an excellent idea, it is a rule that we may easily miss.


> I don't have a strong sense one way or the other, but it's unclear to me that this is something specific to the component, vs. a generalized function which could be defined outside the component definition (also allows for easier unit testing).

I agree with this; the functions were moved outside the component and documentation was added to them.